driver/infoblox_ipam.go

```diff
@@ -56,42 +58,45 @@ func getPrefixLength(cidr string) (prefixLength string) {
 }
 
 func (ibDrv *InfobloxDriver) RequestAddress(r *ipamApi.RequestAddressRequest) (*ipamApi.RequestAddressResponse, error) {
-	network := ibclient.BuildNetworkFromRef(r.PoolID)
+	logrus.Debugf("RequestAddress Called : %+v\n", r)
 
+	network := ibclient.BuildNetworkFromRef(r.PoolID)
 	macAddr := r.Options[netlabel.MacAddress]
-	if len(macAddr) == 0 {
-		macAddr = ibclient.MACADDR_ZERO
-		logrus.Infof("RequestAddressRequest contains empty MAC Address. '%s' will be used.\n", macAddr)
-	}
-
-	fixedAddr, _ := ibDrv.objMgr.GetFixedAddress(network.NetviewName, network.Cidr, "", macAddr)
-	if fixedAddr != nil {
-		if r.Address != "" {
-			if fixedAddr.IPAddress != r.Address {
-				msg := fmt.Sprintf("Requested MAC '%s' is already associated with a difference IP '%s' (requested: '%s')",
-					macAddr, fixedAddr.IPAddress, r.Address)
-				logrus.Infof("RequestAddress: %s", msg)
-				return &ipamApi.RequestAddressResponse{}, errors.New(msg)
-			}
-		}
-	}
-
+	var fixedAddr *ibclient.FixedAddress
 	var err error
-	if fixedAddr == nil {
+	// TODO: Do not create multiple gateways for shared networks
+
+	// Gateway request does not have a MAC address in it. If RequestAddressType is Gateway
+	// then allocate IP with MAC 00:00:00:00:00:00
+	// If RequestAddressType is not Gateway and MAC is empty then its overlay network
+	// request for the Gateway.
+	if rType, ok := r.Options["RequestAddressType"]; macAddr == "" || (ok && rType == netlabel.Gateway) {
```

Comment:
Curious...

So, there are cases where macAddr is "" even when the interface is NOT a gateway?